aten/src/ATen/native/cuda/UnaryOpsKernel.cu

```diff
@@ -25,10 +25,21 @@ void bitwise_not_kernel_cuda(TensorIterator& iter) {
   }
 }
 
+// We manually overload exp because ::exp does not work with thrust::complex types.
+template<typename scalar_t>
+__host__ __device__ static inline scalar_t exp_wrapper(scalar_t v) {
+  return ::exp(v);
+}
+
+template<typename T>
+__host__ __device__ static inline c10::complex<T> exp_wrapper(c10::complex<T> v) {
+  return std::exp(v);
+}
+
 void exp_kernel_cuda(TensorIterator& iter) {
-  AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.dtype(), "exp_cuda", [&]() {
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES_AND1(ScalarType::Half, iter.dtype(), "exp_cuda", [&]() {
     gpu_kernel(iter, []GPU_LAMBDA(scalar_t a) -> scalar_t {
-      return ::exp(a);
+      return exp_wrapper(a);
```

Comment:
Can't we just use `std::complex` here?

Reply:
`std::exp` often doesn't work for `float` and `double` on ROCm. See https://github.com/pytorch/pytorch/blob/1d0ec50a02616d2b1e11628670314b3c6e1ad12d/aten/src/ATen/native/Distributions.h#L7

Reply:
OK, then can you comment on that in the code?

Reply:
Done